pkg/provider/virtualmachine/virtualmachine.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package virtualmachine
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-07-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+type Variant string
```

Comment:
`Variant` is too generic, what about`VirtualMachineType`?

Reply:
The package name indicates the use of it would be `virtualmachine.Variant`. I think it avoids repetition and has the same meaning as `VirtualMachineType`.

Reply:
I'm O.K. with Variant (I agree it's a bit generic but I can't think of anything better!)